### DISCUSSION TITLE
第1节课笔记(25班)

### DISCUSSION THREAD

**Discussion by zcg-joker** [InternLM/Tutorial#199] [笔记]
请上传第1节课的笔记在此处~

**Comment by TZWXBLRS** [comment]
发展方向：由专用模型到通用模型
书生·浦语大模型介绍（1）开源历程
书生·浦语大模型介绍（2）系列
书生·浦语大模型介绍（3）性能
从模型到应用过程

第一步模型选型。针对于应用场景，比较多种大模型相关维度的能力，进行模型评测。经过模型评测初步选型之后，可选定意向大模型。

第二步评估业务场景复杂度。若不复杂，可直接把模型拿过来应用。则继续如果业务场景非常复杂，通常来讲直接开源模型无法满足需求，需要微调、prompt工程等进一步构建。

第三步，判断全参数微调或部分微调等，并计算所需算力，应提前规划，避免微调失败。如果算力足够，可进行全参数微调。如果算力资源比较受限，只能进行部分参数微调，类似把大部分的参数固定住，只调一小部分参数。

第四步，构建基于大模型的智能体需要考虑模型与环境的交互。如果需要调用外部API或与已有业务数据库交互，就需要构建智能体。如果不需要与环境交互，就可以直接将微调好的模型在业务场景中试用。

第五步，模型评测，并评估是否上线应用或继续迭代。

第六步，模型部署。关于软件系统相关性能、安全、功能等方面内容。如考虑如何以更少的资源部署模型，或者如何提升整个应用的吞吐量。

**Reply by zcg-joker** [reply]
nice!但是写成笔记链接的形式会更好哈~

**Comment by sxswz213** [comment]
https://blog.csdn.net/sxswz/article/details/135483000?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135483000%22%2C%22source%22%3A%22sxswz%22%7D

**Reply by zcg-joker** [reply]
nice!

**Comment by LordHidy** [comment]
https://github.com/LordHidy/InternLM_learning/blob/main/Notes/Lesson_1.md

**Reply by zcg-joker** [reply]
nice!

**Comment by qqqmmmxxx** [comment]
https://zqm6a0l4f0l.feishu.cn/wiki/Pv7VwKByIim5OfkPaLScc172ndg?from=from_copylink

**Reply by zcg-joker** [reply]
nice~

**Comment by wunuooo** [comment]
https://blog.csdn.net/ynx005/article/details/135484860?spm=1001.2014.3001.5502

**Reply by zcg-joker** [reply]
nice~

**Comment by liliang1918** [comment]
https://blog.csdn.net/lalala12ll/article/details/135487885?spm=1001.2014.3001.5501

**Reply by zcg-joker** [reply]
nice~

**Comment by SpongeBob-0715** [comment]
https://p065drvsp5m.feishu.cn/docx/SnXldHZkMoeNQLx2F7qcgsRzn1c?from=from_copylink

**Reply by zcg-joker** [reply]
nice~

**Comment by FableFatale** [comment]
https://www.cnblogs.com/yunyetui/p/17955319

**Reply by zcg-joker** [reply]
nice~

**Comment by jhyz** [comment]
https://blog.csdn.net/jinghangyz/article/details/135488141?spm=1001.2014.3001.5502

**Reply by zcg-joker** [reply]
nice~

**Comment by fxlt** [comment]
https://github.com/fxlt/internLM-Study/blob/main/1.%E4%B9%A6%E7%94%9F%E6%B5%A6%E8%AF%AD%E5%A4%A7%E6%A8%A1%E5%9E%8B%E5%85%A8%E9%93%BE%E8%B7%AF%E5%BC%80%E6%BA%90%E4%BD%93%E7%B3%BB.md

**Reply by zcg-joker** [reply]
nice~

**Comment by 12345567890hc** [comment]
https://www.mubu.com/doc/L8ggfGgY3C

**Reply by zcg-joker** [reply]
nice~

**Comment by zengya55** [comment]
https://github.com/zengya55/InterLM_Learning/blob/main/lesson1/InterLM%20course%201.md

**Reply by zcg-joker** [reply]
nice~

**Comment by ronidong** [comment]
https://blog.csdn.net/weixin_52524888/article/details/135491112

**Reply by zcg-joker** [reply]
nice~

**Comment by BaiYu96** [comment]
https://blog.csdn.net/zhh763984017/article/details/135491600

**Reply by zcg-joker** [reply]
nice~

**Comment by juxingyiwan** [comment]
https://github.com/juxingyiwan/InternLM_Camp_202401

**Reply by zcg-joker** [reply]
nice~

**Comment by MQDCY** [comment]
https://blog.csdn.net/weixin_48396757/article/details/135494823?spm=1001.2014.3001.5501

**Reply by zcg-joker** [reply]
nice~

**Comment by ykuang** [comment]
https://github.com/ykuang/internlm-tutorial/blob/main/notes/lesson1.md

**Reply by zcg-joker** [reply]
nice~

**Comment by jimmyrich7203** [comment]
书生·浦语大模型系列

轻量级：InternLM-7B
70亿模型参数
1000亿训练token数据
长语境能力，支持8K语境窗口长度
通用工具调用能力，多种工具调用模板
中量级：InternLM-20B
200亿模型参数，在模型能力与推理代价间取得平衡
采用深而窄的结果，降低推理计算量但提高推理能力
4K训练语境长度，推理时可外推至16K
重量级：1230亿模型参数，强大的性能
极强推理能力、全面的知识覆盖面、超级理解能力与对话能力
准确的API调用能力，可实现各类Agent
从模型到应用流程
- 浦语大模型全链路开源体系生态

数据：书生·万卷
2TB数据，涵盖多种模态与任务
预训练：InternLM-Train
并行训练，极致优化速度达到3600 Tokens/sec/gpu
微调：XTuner
支持全参数微调
支持LoRA等低成本微调
部署：LMDeploy
全链路部署，性能领先
每秒生成2000+tokens
评测：OpenCompass
全方位评测，性能可复现80套评测集，40万道题目
应用：
Legent（轻量级智能体框架）：搭建简单的智能体
AgentLego：支持多种智能体，支持代码解释器等多种工具
全链路开源开放体系|数据

文本数据：50亿个文档，数据量超1TB
图像-文本数据集：超2200万个文件，数据量超140GB
视频数据：超1000个文件，数据量超900GB
特点：
多模态融合：万卷包含文本、图像和视频等多模态数据，涵盖科技、文学、媒体、教育和法律等多个领域。该数据集对模型的知识内容、逻辑推理和泛化能力的提升有显著提升。
精细化处理：万卷经过语言筛选、文本提取、格式标准化、数据过滤和清洗（基于规则和模型）、多尺度去重和数据质量评估等精细数据处理环节，能够很好地适应后续模型训练的要求
价值观对齐：在万卷的构建过程中，研究人员注重将数据内容与主流中国价值观进行对齐，并通过算法和人工评估的结合提高语料库的纯净度。
全链路开源开放体系|预训练

高可扩展：支持从8卡到千卡训练，千卡加速效率达92%。
极致性能优化：Hybrid Zero独特技术+极致优化，加速50%
兼容主流：无缝接入HuggingFace等技术生态，支持各类轻量化技术
开箱即用：支持多种规格语言模型，修改配置即可训练
全链路开源开放体系|微调

增量续训
使用场景：让基座模型学习到一些新知识，如某个垂类领域知识
训练数据：文章、书籍、代码等
有监督微调
使用场景：让模型学会理解和遵循各种指令，或者注入少量领域知识
训练数据：高质量对话、问答数据
高效微调框架：XTuner



全链路开源开放体系|微调


OpenCompass评测体系


全链路开源开放体系|部署

大语言模型特点
内存开销巨大
庞大的参数量
采用自回归生成token,需要缓存k/v
动态Shaps
请求数不固定
token逐个生成，且数量不定
模型结构相对简单
transformer结构，大部分是decoder-only
技术挑战
设备

低存储设备：消费级显卡、移动端等如何部署？
推理

如何加速token的生成速度
如何解决动态shape，让推理可以不间断
如何有效管理和利用内存
服务

提升系统整体吞吐量
降低请求的平均响应时间
部署方案
技术点
模型并行
低比特量化
Atterntion优化
计算和访存优化
Continuous Batching
解决方案-LMDeploy

全链路开源开放体系|智能体


轻量级智能体Lagent

多模态智能体工具箱AgentLego

丰富的工具集合，尤其是提供了大量视觉、多模态相关领域的前沿算法功能
支持多个主流智能体系统，如LangChain,Transformers Agent，Lagent等
灵活的多模态工具调用接口，可以轻松支持各类输入输出格式的工具函数
一键式远程工具部署，轻松使用和调试大模型智能体

**Reply by zcg-joker** [reply]
nice~

**Comment by jimmyrich7203** [comment]
第一节笔记 1609
https://github.com/jimmyrich7203/InternLMlecture/blob/main/%E6%B5%A6%E8%AF%ADLec1.pdf

**Reply by zcg-joker** [reply]
nice~

**Comment by GuoYiFantastic** [comment]
第一节课笔记  sjtu 郭一凡
https://github.com/GuoYiFantastic/InternLM_training_camp/blob/main/%E4%B9%A6%E7%94%9F%C2%B7%E6%B5%A6%E8%AF%AD%E5%A4%A7%E6%A8%A1%E5%9E%8B%E5%85%A8%E9%93%BE%E8%B7%AF%E5%BC%80%E6%BA%90%E4%BD%93%E7%B3%BB/README.md

**Reply by zcg-joker** [reply]
nice~

**Comment by Cyberworldvoyager** [comment]
第一节课笔记 by纪佳翼
https://github.com/Cyberworldvoyager/Interlm_lecture/blob/Cyberworldvoyager-patch-1/%E7%AC%AC%E4%B8%80%E8%8A%82%E8%AF%BE%E7%AC%94%E8%AE%B0%20by%E7%BA%AA%E4%BD%B3%E7%BF%BC.pdf

**Reply by zcg-joker** [reply]
nice~

**Comment by CharlineSu** [comment]
https://cewmxvz60v.feishu.cn/wiki/K5tDwibmwiAsWckr9cpcZAipnwc

**Reply by zcg-joker** [reply]
![image](https://github.com/InternLM/tutorial/assets/77949815/f0fa7057-ed12-423a-aa0c-68afce5e1bc1)
记得打开一下阅读权限哈~ 不然看不到你优秀的笔记了~

**Comment by JJbond-sheng** [comment]
https://blog.csdn.net/2301_80135859/article/details/135540106

**Reply by zcg-joker** [reply]
哇~ 这个手写笔记我觉得也很nice~

**Comment by wade361** [comment]
第一课笔记＿Wade
https://github.com/wade361/-Wade/issues/1

**Reply by zcg-joker** [reply]
不要把讲稿识别出来了当笔记哈~

**Reply by wade361** [reply]
OK

**Comment by Alphonsinaone** [comment]
第一节课笔记
https://github.com/Alphonsinaone/InternLM/blob/main/Study%20Notes/Lesson_1.md

**Reply by zcg-joker** [reply]
nice~

**Comment by IdealDestructor** [comment]
https://blog.deqiang.wang/2024/01/12/Project-LLM1/

**Reply by zcg-joker** [reply]
nice~ 你的博客网站好酷哈哈哈啊~
